test/cpp/ext/otel/otel_plugin_test.cc

```diff
@@ -63,6 +62,18 @@ class MockMetricReader : public opentelemetry::sdk::metrics::MetricReader {
 class OTelPluginEnd2EndTest : public ::testing::Test {
  protected:
   void SetUp() override {
+    // We are resetting the MeterProvider and OpenTelemetry plugin at the start
+    // of each test to avoid test results from one test carrying over to another
+    // test. (Some measurements can get arbitrarily delayed.)
+    auto meter_provider = new opentelemetry::sdk::metrics::MeterProvider;
+    opentelemetry::metrics::Provider::SetMeterProvider(
+        opentelemetry::nostd::shared_ptr<opentelemetry::metrics::MeterProvider>(
+            meter_provider));
+    g_reader_.reset(new grpc::testing::MockMetricReader);
```

Comment:
I think this can use `std::make_shared<>`.

Reply:
opentelemetry has vendored their own shared_ptr implementation `opentelemetry::nostd::shared_ptr` which does not have an automatic conversion from std::shared_ptr :(